public/app/plugins/datasource/prometheus/querybuilder/components/metric-encyclopedia/ResultsTable.tsx

```diff
@@ -0,0 +1,166 @@
+
+  const theme = useTheme2();
+  const styles = getStyles(theme, disableTextWrap);
+
+  const tableRef = useRef<HTMLTableElement | null>(null);
+
+  function isSelectedRow(idx: number): boolean {
+    return idx === selectedIdx;
+  }
+
+  function selectMetric(metric: MetricData) {
+    if (metric.value) {
+      onChange({ ...query, metric: metric.value });
+      reportInteraction('grafana_prom_metric_encycopedia_tracking', {
+        metric: metric.value,
+        hasMetadata: state.hasMetadata,
+        totalMetricCount: state.totalMetricCount,
+        fuzzySearchQuery: state.fuzzySearchQuery,
+        fullMetaSearch: state.fullMetaSearch,
+        selectedTypes: state.selectedTypes,
+        letterSearch: state.letterSearch,
+      });
+      onClose();
+    }
+  }
+
+  useEffect(() => {
+    const tr = tableRef.current?.getElementsByClassName('selected-row')[0];
+    tr?.scrollIntoView({ block: 'nearest', inline: 'nearest' });
+  }, [selectedIdx]);
+
+  function metaHighlighting(metric: MetricData) {
+    if (state.fullMetaSearch && metric) {
+      return (
+        <>
+          <td>
+            <Highlighter
+              textToHighlight={metric.type ?? ''}
+              searchWords={state.metaHaystackMatches}
+              autoEscape
+              highlightClassName={styles.matchHighLight}
+            />
+          </td>
+          <td>
+            <Highlighter
+              textToHighlight={metric.description ?? ''}
+              searchWords={state.metaHaystackMatches}
+              autoEscape
+              highlightClassName={styles.matchHighLight}
+            />
+          </td>
+        </>
+      );
+    } else {
+      return (
+        <>
+          <td>{metric.type ?? ''}</td>
+          <td>{metric.description ?? ''}</td>
+        </>
+      );
+    }
+  }
+
+  return (
+    <table className={styles.table} ref={tableRef}>
+      <thead>
+        <tr className={styles.header}>
+          <th>Name</th>
+          <th>Type</th>
+          <th>Description</th>
+        </tr>
+      </thead>
+      <tbody>
+        <>
+          {metrics &&
+            metrics.map((metric: MetricData, idx: number) => {
+              return (
+                <tr
+                  key={metric?.value ?? idx}
+                  className={`${styles.row} ${isSelectedRow(idx) ? `${styles.selectedRow} selected-row` : ''}`}
+                  onClick={() => selectMetric(metric)}
+                >
+                  <td>
+                    <Highlighter
+                      textToHighlight={metric?.value ?? ''}
+                      searchWords={state.fullMetaSearch ? state.metaHaystackMatches : state.nameHaystackMatches}
+                      autoEscape
+                      highlightClassName={styles.matchHighLight}
+                    />
+                  </td>
+                  {metaHighlighting(metric)}
+                </tr>
+              );
+            })}
+        </>
+        <tr></tr>
```

Comment:
empty row on purpose? Can this be solved with styles instead of markup?

Reply:
Nope, just a mistake 😆